jib-core/src/test/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStepTest.java

```diff
@@ -297,6 +300,25 @@ public void testGetCachedBaseImages_emptyCache()
     Assert.assertEquals(Arrays.asList(), pullBaseImageStep.getCachedBaseImages());
   }
 
+  @Test
+  public void testGetCachedBaseImages_cachedWithoutAllLayers()
+      throws InvalidImageReferenceException, CacheCorruptedException, IOException,
+          LayerCountMismatchException, PlatformNotFoundInBaseImageException,
+          BadContainerConfigurationFormatException, UnlistedPlatformInManifestListException {
+    ImageReference imageReference = ImageReference.parse("cat");
+    Mockito.when(buildContext.getBaseImageConfiguration())
+        .thenReturn(ImageConfiguration.builder(imageReference).build());
+    ManifestTemplate manifest = Mockito.mock(ManifestTemplate.class);
+    ImageMetadataTemplate imageMetadata =
+        new ImageMetadataTemplate(
+            null, Arrays.asList(new ManifestAndConfigTemplate(manifest, null)));
+
+    Mockito.when(cache.retrieveMetadata(imageReference)).thenReturn(Optional.of(imageMetadata));
+    Mockito.when(cache.allLayersCached(manifest)).thenReturn(false);
+
+    Assert.assertEquals(Arrays.asList(), pullBaseImageStep.getCachedBaseImages());
+  }
```

Comment:
Let's use `Truth` for newer assertions. It's a little easier to read since the expected is always on the right side and provides better error messaging. We'll have to add the following import statement:
```
import static com.google.common.truth.Truth.assertThat;
```

Reply:
Updated